### PR TITLE
FOGL-8199: Same datapoints/attribute for different asset not displaying correctly in the graph

### PR DESCRIPTION
**Pull request by mshariq-nerd**

### CODE REVIEW

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 1]
```diff
@@ -492,80 +492,95 @@ export class ReadingsGraphComponent implements OnDestroy {
...
-          allAssetsReading.push(...readings[asset]);
+          let formatedReadings = readings[asset].map(d => { return {reading: d.reading, timestamp: d.timestamp}}).map(r => {
+            r.reading =  Object.assign(
+              {},
+              ...Object.keys(r.reading).map(key => ({[`${asset}.${key}`]: r.reading[key]}))
+            )
+            return r;
+          })
+          allAssetsReading.push(...formatedReadings);
+          assetFormatedReadings = this.readingFormate(allAssetsReading );
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:509]: Format (remove e)

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 2]
```diff
@@ -492,80 +492,95 @@ export class ReadingsGraphComponent implements OnDestroy {
...
-    const imageReadings = [];
-
+   let assetFormatedReadings = {numReadings:[], strReadings:[], arrReadings:[], imageReadings:[]};
     // In case of multiple assets readings, merge all readings
     if ((this.additionalAssets.length > 1 && readings.length !== 0) || !(Array.isArray(readings))) {
       if (Object.keys(readings).length === this.additionalAssets.length) {
         let allAssetsReading = [];
         this.additionalAssets.forEach((asset) => {
-          allAssetsReading.push(...readings[asset]);
+          let formatedReadings = readings[asset].map(d => { return {reading: d.reading, timestamp: d.timestamp}}).map(r => {
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:501]: tt

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 3]
```diff
@@ -492,80 +492,95 @@ export class ReadingsGraphComponent implements OnDestroy {
   }
 
   getReadings(readings: any, optedTime: number) {
-    const numReadings = [];
-    const strReadings = [];
-    const arrReadings = [];
-    const imageReadings = [];
-
+   let assetFormatedReadings = {numReadings:[], strReadings:[], arrReadings:[], imageReadings:[]};
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:495]: readingsClassificationPerType?

Comment (praveen-garg) [reply src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:495]: Where does nested readings go? e.g. {x: {y: 1}, ...}

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 4]
```diff
@@ -492,80 +492,95 @@ export class ReadingsGraphComponent implements OnDestroy {
...
+          key:k,
+          read: { x: r.timestamp, y: value }
+        });
+      } else if (typeof value === 'string') {
+        if (value.includes("__DPIMAGE")) {
+          this.getImageReadingsDimensions(value);
+          imageReadings.push({
+            datapoint:  k,
+            imageData: value,
+            timestamp: this.dateFormatter.transform(r.timestamp, 'YYYY-MM-DD HH:mm:ss.SSS')
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:554]: We should probably define ts_fmt = 'YYYY-MM-DD HH:mm:ss.SSS' as const on top, as this is being used at multiples places

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 5]
```diff
@@ -492,80 +492,95 @@ export class ReadingsGraphComponent implements OnDestroy {
...
+        });
+      } else if (typeof value === 'object') {
+        strReadings.push({
+          key:  k,
+          data: JSON.stringify(value),
+          timestamp: r.timestamp
+        });
+      }
+      else {
+        console.log('Failed to parse reading ', value, ' for key ', k);
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:577]: We should also print type of value

**Review by hemantgarg923 — APPROVED** [review]

**Review by monika-Dianomic — APPROVED** [review]

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**